Use `systemd-modules-load` in place of custom driver-loading unit

LRitzdorf opened the pull request:
**NOTE:** Merge only after #13!

This is here now, in case we want to merge it at some point!

Setup is pretty trivial — the only really new thing here is the need for a `make modules_install` invocation. I've added a Makefile target for that, and it's documented in the associated README as well.

If there's anything else that should go along with this (especially any textbook revision notes), feel free to push more commits to this branch prior to merging! Maintainers should have permission to do that.

Resolves #9

LRitzdorf committed:
modules: replace custom unit with systemd-modules-load

LRitzdorf committed:
modules(make): add modules_install target to Makefiles

tvannoy commented on examples/passthrough/linux/ad1939/Makefile:
```diff
@@ -2,6 +2,9 @@ KDIR ?= ../linux-socfpga
 	$(MAKE) -C $(KDIR) ARCH=arm M=$(CURDIR) CROSS_COMPILE=arm-linux-gnueabihf-
 
+modules_install:
+	$(MAKE) -C $(KDIR) ARCH=arm M=$(CURDIR) modules_install
```
Considering that we already define `KDIR` in the makefiles, I think we should also define `INSTALL_MOD_PATH`, that way we don't have to define the variable every time we call `make`.

However, perhaps we should remove `KDIR` and `INSTALL_MOD_PATH` from the makefiles and just have students export the variables by sourcing a script.

LRitzdorf replied:
Normally I'd agree with the environment variable approach, but there's one issue with this, when used in combination with `sudo` (which is needed in order to edit files in our NFS-shared rootfs). That is, `sudo` strips most/all environment variables from the spawned process (by default; this can be configured). For example:
```sh
$ env | wc -l
85
$ sudo env | wc -l
20
```

We can still add `INSTALL_MOD_PATH` to the Makefile directly, though. I'll commit that shortly.

tvannoy replied:
Yes, good point about `sudo` spawning a new process with different path/variables.

I'd be more inclined towards the environment variable approach in general, as well, but for the students, I think using variables in the makefile is easier (no need to remember the variable names).

tvannoy commented on examples/passthrough/linux/module-loading/README.md:
````diff
@@ -0,0 +1,28 @@
+- Install the module to your network-shared root filesystem
+  - From the same directory, run this command:
+    ```sh
+    sudo KDIR=<path_to_linux-socfpga_repo> make INSTALL_MOD_PATH=/srv/nfs/de10nano/ubuntu-rootfs modules_install
````
Defining the `KDIR` variable here is not necessary because it is currently defined in the makefile.

LRitzdorf replied:
Good point, thanks! I did that on my own system, since my `KDIR` isn't where the Makefile defaults to, and blindly copied the command from my shell history. Will fix shortly.

LRitzdorf committed:
modules(make): specify default KDIR in Makefiles

LRitzdorf committed:
modules(make): compose make command once, for clarity

Instead of defining all of our options for every target, let's just do
it once, at the top of the Makefile.

tvannoy commented on examples/passthrough/linux/module-loading/README.md:
```diff
@@ -0,0 +1,28 @@
+- Compile your kernel module as usual
+  - Run the `make` command from the module's directory (for example, the [`ad1939`](../ad1939) directory)
+- Install the module to your network-shared root filesystem
+  - From the same directory, run this command:
```
I think this is more clear. We have to be explicit about paths for people who are learning.
```suggestion
  - From the module's directory, run this command:
```

tvannoy commented on examples/passthrough/linux/module-loading/README.md:
```diff
@@ -0,0 +1,28 @@
+
+## For Regular Use
+
+In order to load modules on-demand, we can take advantage of automatic loading functionality that's built into Linux (or more precisely, Systemd).
```
While technically we could use the systemd service to load "on-demand" anytime, the intended and normal use is loading at boot time.
```suggestion
In order to load modules on boot, we can take advantage of [automatic module loading functionality that's built into systemd](https://www.freedesktop.org/software/systemd/man/latest/modules-load.d.html).
```

Also, if we're going to be pedantic, we should include links to the documentation, which I've added...

LRitzdorf committed:
modules(readme): accept suggestions from code review

Co-authored-by: Trevor Vannoy <tvannoy+github@pm.me>

tvannoy commented:
I will test this when I get the chance. I don't have an NFS rootfs set up, nor do I plan on making one, but I will change the install path accordingly.

LRitzdorf commented:
Please do! Also, for what it's worth, I've been using exactly this system since last semester 🙂

tvannoy commented:
Well yeah, I don't doubt that it works, just want to verify as a matter of principle before merging.

LRitzdorf committed:
modules(make): actually specify INSTALL_MOD_PATH

LRitzdorf committed:
modules(readme): missing-file warnings are fine

These occur if the user hasn't run `make modules_install` from the main
kernel repo, and as such, files like `modules.deps` don't exist.
However, this doesn't prevent our custom modules from being loaded
successfully.

LRitzdorf committed:
devtree: place TPA6130A2 node underneath I2C bus

This properly reflects the system's physical and logical layout.

LRitzdorf committed:
modules(tpa6130a2): add modern driver, based on AD1939

LRitzdorf committed:
modules(tpa613a2): remove now-obsolete amp driver

This one was also mis-named, which was confusing at first.

LRitzdorf committed:
systemd: update module-load script with new `tpa6130a2` name

LRitzdorf commented:
Let's delay this until #13 is merged, since its naming changes affect this PR.

LRitzdorf commented:
Just rebased on top of #13, so this is **to be merged only after that**.
